jcheck/src/main/java/org/openjdk/skara/jcheck/BinaryCheck.java

```diff
@@ -23,33 +23,85 @@
 package org.openjdk.skara.jcheck;
 
 import org.openjdk.skara.census.Census;
+import org.openjdk.skara.vcs.BinaryPatch;
 import org.openjdk.skara.vcs.Commit;
 import org.openjdk.skara.vcs.openjdk.CommitMessage;
 
+import java.io.IOException;
+import java.nio.file.Files;
 import java.util.Iterator;
 import java.util.ArrayList;
+import java.util.Map;
 import java.util.logging.Logger;
+import java.util.regex.Pattern;
 
 public class BinaryCheck extends CommitCheck {
     private final Logger log = Logger.getLogger("org.openjdk.skara.jcheck.binary");
 
     @Override
     Iterator<Issue> check(Commit commit, CommitMessage message, JCheckConfiguration conf, Census census) {
         var metadata = CommitIssue.metadata(commit, message, conf, this);
+        var fileSizeLimits = conf.checks().binary().fileSizeLimits();
 
         var issues = new ArrayList<Issue>();
         for (var diff : commit.parentDiffs()) {
             for (var patch : diff.patches()) {
-                if (patch.isBinary() &&
-                    (patch.status().isAdded() || patch.status().isCopied())) {
-                    issues.add(new BinaryIssue(patch.target().path().get(), metadata));
+                if (patch.isTextual()) {
+                    continue;
+                }
+                // Here, the file is surely binary.
+                var binaryPatch = (BinaryPatch) patch;
+                var path =  binaryPatch.target().path().orElse(null);
+                var fileName = path != null ? path.getFileName().toString() : "";
+
+                long fileSize = 0;
+                boolean needCheck = false;
+
+                if (binaryPatch.status().isAdded()) {
+                    // This is a new-added file, so the BinaryHunk#inflatedSize of the first hunk should be the file size.
+                    fileSize = binaryPatch.hunks().get(0).inflatedSize();
+                    needCheck = true;
+                } else if (binaryPatch.status().isRenamed() || binaryPatch.status().isCopied()
+                        || binaryPatch.status().isModified() || binaryPatch.status().isUnmerged()) {
+                    // Use the file size in the file system.
+                    try {
+                        fileSize = path != null ? Files.size(path) : 0;
+                        needCheck = true;
```

Comment:
Does that mean that jcheck will block renaming an existing binary file if it is larger than the currently accepted maximum? I don't think that is the right behavior. Only adding a new file should trigger this.

Reply:
I intended to check it. But if you think it don't need to check, I could remove it.